src/server/worker/datum/datum_v2.go

```diff
@@ -0,0 +1,364 @@
+package datum
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"time"
+
+	"github.com/gogo/protobuf/jsonpb"
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/pps"
+	"github.com/pachyderm/pachyderm/src/server/pkg/backoff"
+	"github.com/pachyderm/pachyderm/src/server/pkg/sync"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tar"
+	"github.com/pachyderm/pachyderm/src/server/pkg/tarutil"
+	"github.com/pachyderm/pachyderm/src/server/worker/common"
+)
+
+const (
+	// MetaPrefix is the prefix for the meta path.
+	MetaPrefix = "meta"
+	// MetaFileName is the name of the meta file.
+	MetaFileName = "meta"
+	// PFSPrefix is the prefix for the pfs path.
+	PFSPrefix = "pfs"
+	// OutputPrefix is the prefix for the output path.
+	OutputPrefix = "out"
+	// TmpFileName is the name of the tmp file.
+	TmpFileName       = "tmp"
+	defaultNumRetries = 3
+)
+
+// PutTarClient is the standard interface for a client that implements PutTar.
+type PutTarClient interface {
+	// PutTar puts a tar stream.
+	PutTar(r io.Reader, overwrite bool, datum ...string) error
+}
+
+const defaultDatumsPerSet = 10
+
+// SetSpec specifies criteria for creating datum sets.
+type SetSpec struct {
```

Comment:
Is there really that big of a chance that another parameter will be added to this?

Reply:
Yes, there is a configuration in V1 for creating datum sets (they are called chunks in V1) based on file sizes. There will likely be other types of configuration for datum set splitting that we will want in the future.